README.md

```diff
@@ -1,4 +1,223 @@
+        logger: SideEffectLogger
+    ): Job = launch(context = CoroutineName(name)) {
+        for (inputAction in input) {
+            logger.logSideEffectEvent { LogEvent.SideEffectEvent.InputAction(name, inputAction) }
+            if (inputAction is CalculatorAction.Add &&
+                stateAccessor() >= 0) {
+                launch {
+                    val response = makeNetworkCall()
+                    logger.logSideEffectEvent {
+                        LogEvent.SideEffectEvent.Custom(name, "Received network response: $response")
+                    }
+                    if (response == 200) {
+                        val outputAction = CalculatorAction.Deduct(1)
+                        logger.logSideEffectEvent { LogEvent.SideEffectEvent.DispatchingToReducer(name, outputAction) }
+                        output.send(outputAction)
+                    }
+                }
+            }
+        }
+    }
+}
+```
+
+You must register you `sideEffect` inside `createStore(sideEffects = listOf(sideEffect))`:
+```kotlin
+val storeWithSideEffect = coroutineScope.createStore<Int, CalculatorAction>(
+    name = "Calculator",
+    initialState = 0,
+    sideEffects = listOf(sideEffect),
+    reducer = { currentState, newAction ->
+        when (newAction) {
+            is CalculatorAction.Add -> currentState + newAction.value
+            is CalculatorAction.Deduct -> currentState - newAction.value
+            is CalculatorAction.Multiply -> currentState * newAction.value
+            is CalculatorAction.Divide -> currentState / newAction.value
+        }
+    }
+)
+```
+
+When we subscribe to `storeWithSideEffect` and trigger side effect by dispatching right actions -
+network request from side effect will be performed:
+```kotlin
+storeWithSideEffect.subscribe { state -> updateUI(state) }
+storeWithSideEffect.dispatch(CalculatorAction.Deduct(1))
+// Current state will become -1
+storeWithSideEffect.dispatch(CalculatorAction.Add(11))
+// Current state will become 10 and network request will happen
+```
+
+`CoRedux` also provides two simplified implementations of `SideEffect` that you might find useful:
+- `SimpleSideEffect` that produces on one input action either one output action or no action.
+- `CancellableSideEffect` that cancels previously running `Job` and starts a new one if the same type of action is dispatched to the store.
+
+### Implementation details
+
+With default values in `createStore()`, CoRedux starts main reducer coroutine ("manager"), when first `StateReceiver` subscribes to `CoreduxStore` instance.
+"Manager" coroutine is a single source of current state that is defined in local scope of coroutine - this allows to prevent any concurrency problems.
+Then "manager" coroutine itself sequentially:
+- starts all side effects coroutines ("workers"), that listens for incoming actions
+- sets current state to initial state
+- immediately pushes the initial state to the `StateReciever`
```

Comment:
Again, somehow related to 

> With default parameters (launchMode = CoroutineStart.LAZY), on first subscription, store will emit initialState to it.

It might confuses the reader or at least make him / her wonder if it is the same thing or not

Reply:
Yes, seems this need clarification - will add.